TeamCode/src/main/java/org/firstinspires/ftc/teamcode/DriveTrain.java

```diff
@@ -0,0 +1,651 @@
+// SETUP V
+    private String LeftFront = "leftfront";
+    private String RightFront = "rightfront";
+    private String LeftRear = "leftrear";
+    private String RightRear = "rightrear";
+    private final int gyroPort = 3;
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// MISS V
+    private Telemetry telemetry;        //do we need this?
+    //object for reference (telemetry)
+    private boolean encodersCanRun;
+    private double goalEncoderPosition;
+    private double goalBackwardPosition;
+    private double goalLeftPosition;
+    private double goalRightPosition;
+    private double goalRightStrafePosition;
+    private double goalLeftStrafePosition;
+
+    private double testLiftPosition = 1;
+
+    private double currentGyro;
+
+    private double power;
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// CONSTRUCT
+    //calls the second constructor of DriveTrain and passes a reference to the hardware map, telemetry, the 4 string names of the motors in the order left front, right front, left back, right back and the port reference to the gyro.
+    public DriveTrain(HardwareMap hardwareMap, Telemetry telemetry) {
+        telemetry.addData("DriveTrain Startup", "Beginning");
+        telemetry.update();
+        //setup for all the motors.
+        //setup for all the front motors.
+        this.leftFront = hardwareMap.dcMotor.get(LeftFront);
+        //setup for the left front motor.
+        this.rightFront = hardwareMap.dcMotor.get(RightFront);
+        //setup for the right front motor.
+        //setup for all the back motors.
+        this.leftRear = hardwareMap.dcMotor.get(LeftRear);
+        //setup for the left back motor.
+        this.rightRear = hardwareMap.dcMotor.get(RightRear);
+        //setup for the right back motor.
+
+        leftFront.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        rightFront.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        leftRear.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+        rightRear.setMode(DcMotor.RunMode.RUN_USING_ENCODER);
+
+        //sets all the motors power to zero.
+        //do we need this?
+        this.leftFront.setPower(0);
+        this.rightFront.setPower(0);
+        this.leftRear.setPower(0);
+        this.rightRear.setPower(0);
```

Comment:
#AnswerPls see line 99 "do we need this?"

Reply:
See previous answer.

Reply:
See previous reply :)